crates/polars-core/src/chunked_array/comparison/mod.rs

```diff
@@ -884,31 +884,50 @@ impl Not for BooleanChunked {
 }
 
 impl BooleanChunked {
-    /// Check if all values are `true`
-    pub fn all(&self) -> bool {
-        self.downcast_iter().all(compute::boolean::all)
-    }
-
-    /// Check if any value is `true`
+    /// Returns whether any of the values in the column are `true`.
+    ///
+    /// Null values are ignored.
     pub fn any(&self) -> bool {
         self.downcast_iter().any(compute::boolean::any)
     }
 
-    // Three-valued versions which can return None
-    pub fn all_3val(&self, drop_nulls: bool) -> Option<bool> {
-        if drop_nulls || self.null_count() == 0 {
-            Some(self.all())
-        } else {
-            None
+    /// Returns whether all values in the array are `true`.
+    ///
+    /// Null values are ignored.
+    pub fn all(&self) -> bool {
+        self.downcast_iter().all(compute::boolean::all)
+    }
+
+    /// Returns whether any of the values in the column are `true`.
+    ///
+    /// The output is unknown (`None`) if the array contains any null values and
+    /// no `true` values.
+    pub fn any_kleene(&self) -> Option<bool> {
+        let mut result = Some(false);
+        for arr in self.downcast_iter() {
+            match compute::boolean_kleene::any(arr) {
+                Some(true) => return Some(true),
+                None => result = None,
+                _ => (),
+            };
         }
+        result
     }
-    pub fn any_3val(&self, drop_nulls: bool) -> Option<bool> {
-        let res = self.any();
-        if drop_nulls || res {
-            Some(res)
-        } else {
-            None
+
+    /// Returns whether all values in the column are `true`.
+    ///
+    /// The output is unknown (`None`) if the array contains any null values and
+    /// no `false` values.
+    pub fn all_kleene(&self) -> Option<bool> {
+        let mut result = Some(true);
+        for arr in self.downcast_iter() {
+            match compute::boolean_kleene::all(arr) {
+                Some(false) => return Some(false),
+                None => result = None,
```

Comment:
This should directly return `None` right? If there is a single null we should return. A later `Some(true)` result should not overwrite the `None` of previous array.

Reply:
We might encounter a `Some(false)` in a later chunk, in which case the final output should be `Some(false)` (this will be returned directly.

Once `result` is `None`, it will stay `None`. Encountering a `Some(true)` does nothing either way (it's the empty match arm).

Reply:
Oh.. My bad. I misread the description. Specifically line 920. Then this one is good to go!